pkg/telemetry/telemetry.go

```diff
@@ -105,29 +106,52 @@ func (c Component) getClusterID(ctx context.Context) (string, error) {
 	return fmt.Sprintf("kube-system:%s", ns.UID), nil
 }
 
+func (c Component) getWorkerGroup(os string, arch string, cpus int64, mem int64, runtime string) string {
```

Comment:
this could return a struct like:

```go
type workerGroup struct {
    OS string
    Arch string,
    CPUs    int64
    Mem     int64
    Runtime string
}
```

this is just basic types so it is comparable and can be used as a map key. 

After this, the `wm` below would change from:

```go
	wm := make(map[string]workerData)
```

to:

```go
	wm := make(map[workerGroup]int)
```

I think this will make it possible to just: `wm[wg]++` without any kind of nil/existence checks.